includes/entry.inc.php

```diff
@@ -70,6 +71,35 @@
 				$rstatus = save_read_status($connid, $user_id, $id);
 			}
 			
+			if (isset($_SESSION[$settings['session_prefix'] . 'user_id']) && intval($entrydata['locked'])==0) {
+				// vote handling
+				$user_id = $_SESSION[$settings['session_prefix'] . 'user_id'];
+        $result = @mysqli_query($connid, "SELECT TRUE AS 'exists' FROM " . $db_settings['userdata_table'] . " WHERE user_id = " . intval($user_id) . " AND voting_allowed >= 1 LIMIT 1") or raise_error('database_error', mysqli_error($connid));
+  			$exists = mysqli_fetch_row($result);
+  			mysqli_free_result($result);
+  			if (isset($exists) && intval($exists) == 1) {
```

Comment:
The query gives you a `TRUE` back (or not). Why do you ask here for an integer? Same for posting.inc.php line 326 ff.